### ISSUE TITLE
Possible bugs in pg_explainer.py

### ISSUE THREAD

**Issue by ZhongLIFR** [mims-harvard/GraphXAI#74]
Thank you again for your great work!
However, I encountered some confusing coding snippets in your implementation in the script, GraphXAI/graphxai/explainers/pg_explainer.py (Lines 119-123):

            if self.explain_graph: 
                h = torch.cat([h1, h2], dim=1)
            else:
                h3 = emb.repeat(h1.shape[0], 1)
                h = torch.cat([h1, h2], dim=1)

I believe that "self.explain_graph = True" corresponds to graph-level explanation, while "self.explain_graph = False" corresponds to node-level explanation. However, in your implementation (I know that you adapted this from the DIG library), the `h3 = emb.repeat(h1.shape[0], 1)` is defined but never used. So how to distinguish graph-level explanation and node-level explanation?

**Comment by owencqueen** [comment]
Hello, thanks for bringing this to our attention!

As you have pointed out, `h3` is not used, and those lines of code are very redundant. The difference with the option `self.explain_graph` is actually handled upstream of the `__emb_to_edge_mask` function in which you are referring to these lines of code. If you look [here](https://github.com/mims-harvard/GraphXAI/blob/a11e65ffbc4df737f35522a8accf2283a8aeaa37/graphxai/explainers/pg_explainer.py#L226) (where `self.explain_graph==True`) and then [here](https://github.com/mims-harvard/GraphXAI/blob/a11e65ffbc4df737f35522a8accf2283a8aeaa37/graphxai/explainers/pg_explainer.py#L281) (`where self.explain_graph==False`), you see that the edge index fed to the function is different, i.e., it’s the full edge index in the former line and the subsetted edge index w.r.t. the given node in the latter line. This is where the graph vs. node differentiation is made in our implementation. However, the code you've pointed out is redundant; we apologize for this confusion, and we will remove this in a future push to the repository.

Please let us know if you have any other issues with the PGExplainer functionality or anything else in the library. Thanks again!

- Owen and GraphXAI team
